drivers/SmartThings/zigbee-power-meter/src/ezex/init.lua

```diff
@@ -0,0 +1,79 @@
+-- Copyright 2022 SmartThings
+--
+-- Licensed under the Apache License, Version 2.0 (the "License");
+-- you may not use this file except in compliance with the License.
+-- You may obtain a copy of the License at
+--
+--     http://www.apache.org/licenses/LICENSE-2.0
+--
+-- Unless required by applicable law or agreed to in writing, software
+-- distributed under the License is distributed on an "AS IS" BASIS,
+-- WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+-- See the License for the specific language governing permissions and
+-- limitations under the License.
+
+local capabilities = require "st.capabilities"
+local ZigbeeDriver = require "st.zigbee"
+local defaults = require "st.zigbee.defaults"
+local constants = require "st.zigbee.constants"
+local clusters = require "st.zigbee.zcl.clusters"
+local ElectricalMeasurement = clusters.ElectricalMeasurement
+local SimpleMetering = clusters.SimpleMetering
+
+local ZIGBEE_POWER_METER_FINGERPRINTS = {
+  { model = "E240-KR080Z0-HA" }
+}
+
+local is_ezex_power_meter = function(opts, driver, device)
+  for _, fingerprint in ipairs(ZIGBEE_POWER_METER_FINGERPRINTS) do
+      if device:get_model() == fingerprint.model then
+          return true
+      end
+  end
+
+  return false
+end
+
+local do_configure = function(self, device)
+  device:refresh()
+  device:configure()
+end
+
+local device_init = function(self, device)
+  device:set_field(constants.SIMPLE_METERING_DIVISOR_KEY, 1000000, {persist = true})
+  device:set_field(constants.ELECTRICAL_MEASUREMENT_DIVISOR_KEY, 10, {persist = true})
+end
+
+local function energy_meter_handler(driver, device, value, zb_rx)
+  local raw_value_miliwatts = value.value
+  local raw_value_watts = raw_value_miliwatts / 1000
+  local delta_energy = 0.0
+  local current_power_consumption = device:get_latest_state("main", capabilities.powerConsumptionReport.ID, capabilities.powerConsumptionReport.powerConsumption.NAME)
+  if current_power_consumption ~= nil then
+    delta_energy = math.max(raw_value_watts - current_power_consumption.energy, 0.0)
+  end
+  device:emit_event(capabilities.powerConsumptionReport.powerConsumption({energy = raw_value_watts, deltaEnergy = delta_energy })) -- the unit of these values should be 'Wh'
+
+  local multiplier = device:get_field(constants.SIMPLE_METERING_MULTIPLIER_KEY) or 1
+  local divisor = device:get_field(constants.SIMPLE_METERING_DIVISOR_KEY) or 1000000
+  local converted_value = raw_value_miliwatts * multiplier/divisor -- unit: kWh
+  device:emit_event(capabilities.energyMeter.energy({value = converted_value, unit = "kWh"}))
+end
```

Comment:
Please refactor/re-nest the sub-drivers that support this capability so that this code is not duplicated.

A useful thing to do might be to make a powerConsumptionReport sub-handler that, for it's `can_handle` function checks the device profile to see if it supports that capability, rather than using mfr fingerprints.

Reply:
This handler is only for EZEX and It is already overrided from energyMeter_defaut because of powerConsumptionReport capability. Each sensors have different input value unit(W,mW) and different divisor and so on. so I think it is okay to separate their code for maintenance.